client/web/src/site-admin/notifications/NotificationsNavbarItem.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useMemo } from 'react'
+
+import { mdiBellAlert, mdiBellBadge, mdiBellCheck, mdiCheckCircle, mdiInformation } from '@mdi/js'
+import classNames from 'classnames'
+import { groupBy, startCase } from 'lodash'
+
+import { renderMarkdown } from '@sourcegraph/common'
+import { Icon, Badge, Popover, PopoverTrigger, PopoverContent, Text, Button, PopoverTail } from '@sourcegraph/wildcard'
+
+import { AlertType } from '../../graphql-operations'
+
+import { useNotifications } from './hooks/useNotifications'
+
+import styles from './NotificationsNavbarItem.module.scss'
+
+const NotificationsGroup: React.FC<{ name: string; items: { message: string; type: AlertType }[] }> = ({
+    name,
+    items,
+}) => {
+    const hasError = items.find(item => item.type === AlertType.ERROR)
+    const hasWarning = items.find(item => item.type === AlertType.WARNING)
+    const hasInfo = items.find(item => item.type === AlertType.INFO)
+    return (
+        <div>
+            <div className="mb-2">
+                <Icon
+                    svgPath={hasError || hasWarning || hasInfo ? mdiInformation : mdiCheckCircle}
+                    aria-hidden={true}
+                    className={classNames(
+                        styles.icon,
+                        hasError ? styles.danger : hasWarning ? styles.warning : hasInfo ? styles.info : styles.success
+                    )}
```

Comment:
```suggestion
                    className={classNames(
                        styles.icon,
                        [styles.danger]: hasError,
                        [styles.warning]: hasWarning,
                        [styles.info]: hasInfo
                        [styles.success]: !(hasError || hasWarning || hasInfo)
                    )}
```

This is easier to understand and read. Nested ternaries are hard to read sometimes, it'll be nice to make use of `if` statements by maybe abstracting the use in other places to a function.

Reply:
I understand that.

It took me about 2-3 minutes to understand the ternary logic, and I fear it's easy to make a mistake, whereas the object key approach is explicit. 
At first glance, I already understand the snippet you shared. I'd go for repeating it it improves readability and maintainability In my opinion.

Reply:
I see, updated.